activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1830,5 +1837,14 @@ def structurally_incompatible_values_for(other)
           v1 == v2
         end
       end
+
+      def field_ordered_value(column, values)
+        node = Arel::Nodes::Case.new
+        values.each.with_index(1) do |value, order|
+          node.when(column.eq(value)).then(order)
+        end
+
+        Arel::Nodes::Ascending.new(node.else(values.length + 1))
+      end
```

Comment:
Since this is no longer related to `ORDER BY FIELD`, perhaps we should rename this to `case_ordered_value`?  Or, to be more descriptive (and match the other `build_*` methods), maybe `build_case_for_value_position`?

Also, what about bumping this up closer to the [other `order`-related methods](https://github.com/rails/rails/blob/cb5765a12751ca03866a89cf19fefcbc2d93a8ea/activerecord/lib/active_record/relation/query_methods.rb#L1683-L1763)?